test/date_time_parser_test.exs

```diff
@@ -439,6 +467,20 @@ defmodule DateTimeParserTest do
     test_time_parsing("9999999999.999999", ~T[17:46:39.999999])
     test_time_parsing("9999999999.9999999999", ~T[17:46:39.999999])
     test_time_parsing("0000000000", ~T[00:00:00])
+    test_time_parsing("-9999999999.9999999999", ~T[06:13:20.000001])
```

Comment:
Originally, I had all of these as a part of testing `DateTime`, but since they are more focused on time, have them here